src/test/java/nl/dtls/fairdatapoint/database/rdf/repository/catalog/CatalogMetadataRepositoryTest.java

```diff
@@ -0,0 +1,109 @@
+ * THE SOFTWARE.
+ */
+package nl.dtls.fairdatapoint.database.rdf.repository.catalog;
+
+import nl.dtls.fairdatapoint.database.rdf.repository.common.MetadataRepositoryException;
+import nl.dtls.fairdatapoint.utils.MetadataFixtureFilesHelper;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.impl.SimpleValueFactory;
+import org.eclipse.rdf4j.query.TupleQuery;
+import org.eclipse.rdf4j.query.TupleQueryResult;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.cache.Cache;
+import org.springframework.cache.concurrent.ConcurrentMapCacheManager;
+
+import java.util.Collections;
+import java.util.List;
+
+import static nl.dtls.fairdatapoint.config.CacheConfig.DATASET_THEMES_OF_CATALOG_CACHE;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class CatalogMetadataRepositoryTest {
+
+    private static final ValueFactory VALUE_FACTORY = SimpleValueFactory.getInstance();
+
+    private IRI catalogUri = VALUE_FACTORY.createIRI(MetadataFixtureFilesHelper.CATALOG_URI);
+
+    @Mock
+    private Cache cache;
+
+    @Mock
+    private Repository repository;
+
+    @Mock
+    private RepositoryConnection repositoryConnection;
+
+    @Mock
+    private TupleQuery tupleQuery;
+
+    @Mock
+    private TupleQueryResult tupleQueryResult;
+
+    @Mock
+    private ConcurrentMapCacheManager cacheManager;
+
+    @Spy
+    @InjectMocks
+    private CatalogMetadataRepositoryImpl catalogMetadataRepository;
+
+    @Test
+    @DisplayName("Themes for catalog are in cache (no query to triple store)")
+    public void themesInCache() throws MetadataRepositoryException {
+        // GIVEN:
+        when(cacheManager.getCache(DATASET_THEMES_OF_CATALOG_CACHE)).thenReturn(cache);
+        when(cache.get(catalogUri.toString(), List.class)).thenReturn(Collections.emptyList());
+
+        // WHEN:
+        catalogMetadataRepository.getDatasetThemesForCatalog(catalogUri);
+
+        // THEN:
+        verify(catalogMetadataRepository, never()).runSparqlQuery(any(), any(), any());
+    }
+
+    @Test
+    @DisplayName("Themes for catalog are not in cache (we have to query to triple store)")
+    public void themesNotInCache() throws MetadataRepositoryException {
+        // GIVEN:
+        when(cacheManager.getCache(DATASET_THEMES_OF_CATALOG_CACHE)).thenReturn(cache);
+        when(repository.getConnection()).thenReturn(repositoryConnection);
+        when(repositoryConnection.prepareTupleQuery(any())).thenReturn(tupleQuery);
+        when(tupleQuery.evaluate()).thenReturn(tupleQueryResult);
```

Comment:
What's your opinion on using an in-memory store (using `new SailRepository(new MemoryStore())` and `initialize()` on it, 2 LOC) vs mocking all underlying components (11 LOC including the class member declarations)?

Reply:
I will try it ;-)